server/routes/categories.js

```diff
@@ -20,7 +21,14 @@ async function handleCategory(req, res) {
   // FIXME: consider putting this in middleware and save on req
   const {meta, parent, data, root} = await parseUrl(req.path)
 
-  if (!meta || !data) return 'next'
+  if (!meta || !data) { // if no data is returned, check if a redirect exists
+    const pathCache = await cache.get(req.path)
+    if (!pathCache) return 'next'
+
+    const {content} = pathCache
+    if (content.redirect) return res.redirect(content.redirect)
```

Comment:
This logic seems like the right behavior for our middleware layer, but I had originally been thinking we would want to invoke it as a new piece of middleware, potentially injected here: https://github.com/nytimes/library/blob/master/server/index.js#L79

We have a couple different pieces of middleware currently, so I think we'd want to make sure the redirect is only invoked as the last possible step prior to sending a 404. This location is near to that, but could theoretically circumvent a playlist page from serving if a redirect also existed at that location.